src/Transport/Sending/IBatcher.cs

```diff
@@ -1,10 +1,11 @@
 namespace NServiceBus.Transport.AzureServiceBus
 {
     using System.Collections.Generic;
+    using System.Threading.Tasks;
     using Transport;
 
     interface IBatcherInternal
     {
-        IList<BatchInternal> ToBatches(TransportOperations operations);
+        Task<List<BatchInternal>> ToBatches(TransportOperations operations);
```

Comment:
@yvesgoeleven I don't see how we'd call it before.
Fist, only if installers are enabled, we will need to check if manage rights check was executed.
Manage rights check is performed by queue creator and then section manager has to provide section for topology creator to operate on. So we don't have a place where we'd do it up-front.

- TransportResourcesCreator : ICreateQueues
  - AssertManageRights()
  - DetermineResourcesToCreate(queueBindings)
       _if no topics found in section manager, call_ BuildTopicBundles()

Also, `BuildTopicBundles()` can be invoked from 2 additional places:

From the subscription manager

- SubscriptionManager : IManageSubscriptions
  - DetermineResourcesToSubscribeTo
    - BuildSubscriptionHierarchy(eventType)
           _if no topics found in section manager, call_ BuildTopicBundles()

and from the batcher when messages are dispatched

- DeterminePublishDestination(eventType)
  - Batcher.AddMulticastOperationBatches()
      _if no topics found in section manager, call_ BuildTopicBundles()

In all 3 cases section manager has to have the information about _actual_ number of topics.

If you have a suggestion how to tackle it, by all means. I don't have a better idea.